File: pset1/mario_more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // defention
    int n, i, j, k, counter = 0 ;


    // Get the size of the pyramids
    do
    {
        n  = get_int(" Height: ");
    }
    while (n > 8 || n < 1);



    for (i = 1 ; i <= n ; i++)
    {
        // Loop for the left side pyramid

        for (j = n - 1 ; j >= i ; j--)
        {
            printf(" ");
            counter++;
        }

        // get the hash for the left side pyramid

        for (k = n - counter ; k >= 1  ; k--)
        {
            printf("#");
        }

        // space between two pyramids

        printf("  ");

        // loop for right pyramid

        for (int s = 1 ; s <= i ; s++)
        {
            printf("#");

        }

        counter = 0 ;
        printf("\n");
    }

}

File: pset6/mario/more/mario.py

from cs50 import get_int
# Get the height :
h = get_int("Height:")
# Check if the height is in range:
while h < 1 or h > 8:
    print("INVALID")
    h = get_int("Height:")
# Print the pyramid
for i in range(1, h + 1):
    for j in range(1, h + 1):
        # Print the left pyramid
        if j > (h - i):
            print("#", end="")
        else:
            print(" ", end="")
        if j == h:
            print(" ", "#" * i, end="")
    print()
